src/lib/structures/objective.ml

```diff
@@ -0,0 +1,118 @@
+(**************************************************************************)
+(*                                                                        *)
+(*     Alt-Ergo: The SMT Solver For Software Verification                 *)
+(*     Copyright (C) 2013-2023 --- OCamlPro SAS                           *)
+(*                                                                        *)
+(*     This file is distributed under the terms of OCamlPro               *)
+(*     Non-Commercial Purpose License, version 1.                         *)
+(*                                                                        *)
+(*     As an exception, Alt-Ergo Club members at the Gold level can       *)
+(*     use this file under the terms of the Apache Software License       *)
+(*     version 2.0.                                                       *)
+(*                                                                        *)
+(*     ---------------------------------------------------------------    *)
+(*                                                                        *)
+(*     The Alt-Ergo theorem prover                                        *)
+(*                                                                        *)
+(*     Sylvain Conchon, Evelyne Contejean, Francois Bobot                 *)
+(*     Mohamed Iguernelala, Stephane Lescuyer, Alain Mebsout              *)
+(*                                                                        *)
+(*     CNRS - INRIA - Universite Paris Sud                                *)
+(*                                                                        *)
+(*     Until 2013, some parts of this code were released under            *)
+(*     the Apache Software License version 2.0.                           *)
+(*                                                                        *)
+(*     ---------------------------------------------------------------    *)
+(*                                                                        *)
+(*     More details can be found in the directory licenses/               *)
+(*                                                                        *)
+(**************************************************************************)
+
+module Function = struct
+  type t = {
+    e : Expr.t;
+    to_max : bool;
+  }
+
+  let equal { e = e1; _ } { e = e2; _ } = Expr.equal e1 e2
```

Comment:
I think this should also compare `to_max`, otherwise if I do `(minimize x)(maximize x)` it has the same effect as `(maximize x)` whereas it should have the same effect as `(minimize x)` (because we must first minimize x *then* maximize it, which does nothing).

Reply:
For the input file:
```
(set-logic ALL)
(set-option :produce-models true)
(declare-const x Int)
(assert (<= x 10))
(minimize x)
(maximize x)
(check-sat)
(get-model)
(get-objectives)
```
I got
```
unknown
(
  (define-fun x () Int 0)
)
(objectives 
  (x -oo)
  (x (interval (- oo) (+ oo)))
)
```
It's consistent but the user cannot understand there are not the same objective in the output of `get-objectives`.

Reply:
Ah right, this is handled by the order which will be different.

I think that output is fine, by the way — the user can understand that these are not the same objective because there are two entries, and the goal is that `get-objectives` return one entry per `(minimize)` or `(maximize)` call that was made. And the problem is ill-specified anyways, so. FWIW, Z3 gives a similar output on this problem.

(The intended use case is that there will be a tool that will parse this output and match the list in order with the `minimize` and `maximize` calls)